src/main/java/spoon/reflect/visitor/CtScanner.java

```diff
@@ -573,10 +574,10 @@ public void visitCtIf(final CtIf ifElement) {
 	public <T> void visitCtNewClass(final CtNewClass<T> newClass) {
 		enter(newClass);
 		scan(newClass.getAnnotations());
-		scan(newClass.getType());
```

Comment:
@monperrus because ctNewClass.getType() is not a "true" property of a ctNewClass, it delegates to getExecutable().getType(). In this case scanner shouldn't visit same node twice. Also see that CtNewClass extends CtConstructorCall, which have the same logic for getType(), but doesn't have scan(getType()) in it's visit method in CtScanner.